wireworld.py

```diff
@@ -0,0 +1,154 @@
+import json
+import sys
+import copy
+
+import numpy as np
+
+initial_state = [
+[0, 0, 0, 3, 0, 0, 0], 
+[0, 3, 0, 3, 0, 3, 0], 
+[3, 0, 3, 0, 3, 0, 3], 
+[1, 0, 0, 3, 0, 0, 1], 
+[2, 0, 3, 3, 3, 0, 2], 
+[0, 3, 0, 3, 0, 3, 0], 
+[0, 0, 0, 3, 0, 0, 0]
+]
+
+
+def config(initial_state):
+	print("To load a file, type in the filename. To use the default state, press enter")
+	filename = input()
+	if filename[-5:] != ".json":
+		if filename != "":
+			print(filename)
+			raise Exception("Config file is not a .json file")
+	if filename != "":
+		print("yes")
+		with open(filename, "r") as j:
+			initial_state = json.load(j)
+			
+	rowsno = (len(initial_state))     # y coord
+	colsno = (len(initial_state[0]))  # x coord
+	return initial_state, rowsno, colsno
+	
+def arrayprint(arr):
+	for row in arr:
+		for col in row:
+		    print(str(col) + " ", end='')
+		print()		
+
+#  This is temporary, and just used to make it easier to visualise right now
+def visualarr(printarr, initial_state):
+
+	for y in range(len(printarr)):
+		for x in range(len(printarr[0])):
+		    if initial_state[y][x] == 0:
+		    	printarr[y][x] = (" ")
+		    elif initial_state[y][x] == 1:
+		    	printarr[y][x] = ("■")
+		    elif initial_state[y][x] == 2:
+		    	printarr[y][x] = ("▥")
+		    elif initial_state[y][x] == 3:
+		    	printarr[y][x] = ("□")
+	return printarr
+
+def adjcheck(x, y, initial_state, rowsno, colsno):
+	total = 0
+	if y > 0:  		# y-1 wont be out of bounds
```

Comment:
Similarly to the above point; there's a way to do this which cuts down on code and makes readability easier; both work though, mostly just preference.